spread/runner.go

```diff
@@ -234,17 +234,14 @@ func (r *Runner) loop() (err error) {
 		}
 	}
 
-	for {
-		select {
-		case <-r.done:
-			r.alive--
-			if r.alive > 0 {
-				debugf("Worker terminated. %d still alive.", r.alive)
-				continue
-			}
-			debugf("Worker terminated.")
-			return nil
+	for range r.done {
+		r.alive--
+		if r.alive > 0 {
+			debugf("Worker terminated. %d still alive.", r.alive)
+			continue
 		}
+		debugf("Worker terminated.")
+		return nil
```

Comment:
Hm, looks like this is not ready. Current go complains:
```
./runner.go:246:1: missing return at end of function
```
we could of course just move the reutrn down but i would feel better I think if we had a a unit test here, given the function I'm not sure that is easy to do though :/